clients/nodejs/modules/JsonRpcServer.js

```diff
@@ -103,6 +103,7 @@ class JsonRpcServer {
         this._methods.set('getTransactionByBlockHashAndIndex', this.getTransactionByBlockHashAndIndex.bind(this));
         this._methods.set('getTransactionByBlockNumberAndIndex', this.getTransactionByBlockNumberAndIndex.bind(this));
         this._methods.set('getTransactionByHash', this.getTransactionByHash.bind(this));
+        this._methods.set('getTransactionByHash2', this.getTransactionByHash2.bind(this));
```

Comment:
Why must this be a new method, if you are only _adding_ fields? That is not a breaking change, is it?

Reply:
It's not a breaking change as long as the client library/code supports it ([there are de-serialization libraries](https://www.baeldung.com/jackson-deserialize-json-unknown-properties#1-unrecognizedpropertyexception-on-unknown-fields) that require extra configuration to ignore unknown fields or else they will throw and exception/error out). We can´t change custom client code that works this way (nor is our responsibility), but at least we should check and, if needed, fix this on the [community provided libraries](https://github.com/nimiq-community/), but we didn't want to make this `core-js` change dependent on that (the idea is to update and release `core-js` first and then release a new version of the libraries to support the new RPC methods and to make sure they ignore unknown fields, but this second step may take some time).

For PoS we should make sure to release with some kind of versioning scheme that allows us to improve the API while retaining backward compatibility for people who can't update their clients/libraries at the same time as their nodes.